asyncify.mjs

```diff
@@ -48,10 +36,31 @@ function proxyGet(obj, transform) {
   });
 }
 
-class Asyncify {
-  constructor() {
+export class Asyncify {
+  constructor({ dataBegin, dataEnd } = {}) {
     this.value = undefined;
     this.exports = null;
+
+    if (dataBegin !== undefined) {
+      if (dataEnd === undefined) {
+        throw new Error('Must provide dataEnd if dataBegin is provided')
+      }
+      if (dataEnd < dataBegin) {
+        throw new Error('dataEnd must be >= dataBegin')
```

Comment:
Hm it probably should be `>= dataBegin + 8`, otherwise we can still overflow when reading / writing to `dataStart`.